### PR TITLE
Fix: Updated alot of things

### PR DESCRIPTION
**Pull request by Joniii11**
So basically i enabled strict mode in tsconfig.json and fixed all of the errors that came up.

## `Node.ts`:
- Updated Types in the main class
- Checking if the userId is existing
- Added a ? to prevent crashes if ws is undefined
- instead of using delete keyword we assign now reconnectAttempt null

## `Rest.ts`:
- Added the PartialNull type
- Fixed the types in the interface `playOptions`
- Added some docs to some functions
- added missing types in the return value in functions

## `Connection.ts`:
- Updated the voice types to also include `PartialNull<IVoiceServer>`
- Made `<Connection>.setServersUpdate` use async/await

## `CustomFilters.ts`:
- Reworked some things
- Removed unecessary init variables
- Added async/await to the functions
- Made the Docs a bit clearer for the functions

## `Filters.ts`:
- Made all of the functions use async/await
- Update the init variables and their types

## `Player.ts`:
- Fixed numerous types and functions
- Added some ? to prevent crashes
- Finally made use of isAutoPlay variable
- Converted the function `<Player>.disconnect` from `public` to `protected`

And fixed alot more.

I hope this PR will be accepted since this fixes alot of problems (not all but some atleast).

### CODE REVIEW

File: src/Player/Filters.ts [hunk 1]
```diff
@@ -154,187 +153,185 @@ interface lowPassOptions {
...
      * 
      * @param mix An object that conforms to the ChannelMixOptions type that defines the channel mix to apply
      * @returns 
      */
-    public setChannelMix(mix: channelMixOptions): Filters {
-        this.channelMix = mix || null;
-        this.updateFilters();
+    public async setChannelMix(mix: channelMixOptions): Promise<Filters> {
+        this.channelMix = mix || undefined;
+        await await this.updateFilters();
```
Comment (UnschooledGamer) [inline src/Player/Filters.ts]: Double awaits

**Review by UnschooledGamer — CHANGES_REQUESTED** [review]

File: src/Player/Player.ts [hunk 2]
```diff
@@ -350,23 +354,25 @@ export class Player extends EventEmitter {
    * Disconnects the player from the voice channel.
    * @returns {Promise<Player>} - A Promise that resolves to the Player instance.
    */
-  public async disconnect(): Promise<Player> {
-    if (!this.voiceChannel) return
+  protected async disconnect(): Promise<Player> {
```
Comment (UnschooledGamer) [inline src/Player/Player.ts:357]: Why is this method converted into an `protected` method? This method is similar to the `connect`.

File: src/guild/Track.ts [hunk 3]
```diff
@@ -34,9 +34,6 @@ export class Track {
         this.pluginInfo = data.pluginInfo,
         this.userData = data.userData,
         this.info = {
-            isrc: data.info.isrc || null,
-            uri: data.info.uri || null,
-            artworkUrl: data.info.artworkUrl || null,
```
Comment (UnschooledGamer) [inline src/guild/Track.ts:39]: Why are these removed?

File: src/Player/Player.ts (hunk 2 above)
Comment (Joniii11) [reply src/Player/Player.ts:357]: I think it is not needed. Why does anyone need to disconnect the bot from the vc and still have it a player created on lavalink?

File: src/guild/Track.ts (hunk 3 above)
Comment (Joniii11) [reply src/guild/Track.ts:39]: The destructering is already done, refer to the types they are autmatically null if it does not exist

Comment (UnschooledGamer) [reply src/guild/Track.ts:39]: Oh okay, hoping everything works

File: src/Player/Player.ts (hunk 2 above)
Comment (UnschooledGamer) [reply src/Player/Player.ts:357]: There could be an use case which don't have currently in my mind.

Like disconnecting the bot from VC while still keeping the queue & player in-tech for whatever reason.

**Review by UnschooledGamer — APPROVED** [review]

File: src/guild/Track.ts (hunk 3 above)
Comment (Joniii11) [reply src/guild/Track.ts:39]: As far as i have seen in the docs should these just return null

Comment (UnschooledGamer) [reply src/guild/Track.ts:39]: Even `uri`? That the url of song. `artworkURL` does get returned if Lavalink or if the plugin has it in most cases.
